lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -177,20 +217,20 @@ export default function BasicLtiLaunchApp() {
 
     try {
       await authWindow.current.authorize();
-      await fetchContentUrl();
+      await Promise.all([fetchContentUrl(), fetchGroups()]);
     } finally {
-      // eslint-disable-next-line require-atomic-updates
       authWindow.current = null;
     }
-  }, [authToken, canvas, fetchContentUrl]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [authToken, canvas.authUrl]);
```

Comment:
I have not flushed out a comment for this, but its worth discussing here in the PR. To fix this, it needs both fetchContentUrl and fetchGroups, and then it needs those to be wrapped in useCallback along with handleError with their respective dependencies. Doing all this causes an infinite update loop. 

I do need to take a closer look to see if there is a way to simplify the logic or we just need to force the linter to ignore `react-hooks/exhaustive-deps` and control the flow in a more manual way like this.

Reply:
I couldn't reproduce this in testing today. I did run into an issue with authorization was failing because Sean was testing some stuff (see https://hypothes-is.slack.com/archives/CBN3DGW02/p1588787487144200) but that resolved itself once he fixed the devdata keys.